contrib/pam_zfs_key/pam_zfs_key.c

```diff
@@ -721,49 +755,61 @@ pam_sm_chauthtok(pam_handle_t *pamh, int flags,
 			zfs_key_config_free(&config);
 			return (PAM_SERVICE_ERR);
 		}
-		int key_loaded = is_key_loaded(pamh, dataset);
-		if (key_loaded == -1) {
+		if (!old_token) {
+			pam_syslog(pamh, LOG_ERR,
+			    "old password from PAM stack is null");
 			free(dataset);
 			pam_zfs_free();
 			zfs_key_config_free(&config);
 			return (PAM_SERVICE_ERR);
 		}
-		free(dataset);
-		pam_zfs_free();
-		if (! key_loaded) {
+		if (decrypt_mount(pamh, dataset, old_token->value, B_TRUE) == -1) {
```

Comment:
cstyle is also failing in two places due to lines being too long. Please wrap these.

Reply:
Yeah I saw that… still working on more tests btw.

…is it just me or does checkstyle count a tab as 4/8 characters instead of 1? :/

Reply:
It's not you, it counts a tab as 8.